### PR TITLE
Port pill #4

### PR DESCRIPTION
**Pull request by ankhers**

### CODE REVIEW

**Comment by ankhers** [comment]
I will go through this and #1 to find out where I can use `<command>` and `<filename>` tags tomorrow.

File: pills/04-basics-of-language.xml [hunk 1]
```diff
@@ -4,5 +4,362 @@
...
+  <para>
+    <emphasis role="underline">Important</emphasis>: in Nix, everything is an
+    expression, there are no statements. This is common to many functional
+    languages.
+  </para>
+
+  <para>
+    <emphasis role="underline">Important</emphasis>: values in Nix are
+    immutable.
+  </para>
```
Comment (grahamc) [inline pills/04-basics-of-language.xml:47]: I suspect both these Important's could better be `<note>`s.

File: pills/04-basics-of-language.xml [hunk 2]
```diff
@@ -4,5 +4,362 @@
...
+      We've installed nix-repl in the previous pill. If you didn't,
+      <literal>nix-env -i nix-repl</literal>. The nix-repl syntax is slightly
+      different than nix syntax when it comes to assigning variables, but no
+      worries.  I prefer playing with nix-repl with you before cluttering your
+      mind with more complex expressions.
+    </para>
+
+    <para>
+      Launch nix-repl. First of all, nix supports basic arithmetic operations:
+      +, -, and *. The integer division can be done with builtins.div.
```
Comment (grahamc) [inline pills/04-basics-of-language.xml:62]: I wonder if these symbols would be better as `<token>` or `<function>`, same as other operators below.

File: pills/04-basics-of-language.xml [hunk 3]
```diff
@@ -4,5 +4,362 @@
...
+      it's still possible to fallback to plain strings. Parsing urls and paths
+      are convenient for additional safety.
+    </para>
+  </section>
+
+  <section>
+    <title>Identifier</title>
+
+    <para>
+      Not much to say, except that dash (-) is allowed in identifiers. That's
```
Comment (grahamc) [inline pills/04-basics-of-language.xml:120]: maybe a good spot for a `<code>` (or some other tag, but not sure which one.) and several other places here could be improved with a `<code>`.

**Review by grahamc — APPROVED** [review]
A few minor suggestions, but better than what is there! Merging! If/when you update with the suggestions, please make another PR :)

**Pull request merged**